src/components/TopAppBar/TopAppBar.js

```diff
@@ -35,16 +49,33 @@ class TopAppBar extends Component {
             <Typography variant="h6" color="inherit">
               TasmoCompiler {version}
             </Typography>
-            <Tooltip title={<FormattedMessage id="appBarLangTooltip" />}>
+            <div className={classes.language} role="button" tabIndex={0} aria-controls="langs-menu" aria-haspopup="true" onClick={this.handleOpen} onKeyPress={this.handleOpen}>
               <Typography
                 variant="caption"
                 color="inherit"
                 className={classes.language}
               >
                 <LanguageIcon className={classes.leftIcon} />
-                <FormattedMessage id="appBarLangFlag" />
+                <FormattedMessage id="appBarLang" />
               </Typography>
-            </Tooltip>
+            </div>
+            <Menu
+              id="langs-menu"
+              anchorEl={anchorEl}
+              open={Boolean(anchorEl)}
+              onClose={this.handleClose}
+            >
+              {Object.keys(allMessages)
+                .sort()
+                .map((lang) => {
+                  return (
+                    <MenuItem onClick={this.handleClose} id={lang} selected={locale === lang}>
```

Comment:
There has to be a `key` property. I'm not sure that `id` is needed.

Reply:
Add key property.
`id` was there, to use in `handleClose` to pass towards for `changeLanguage`, but now it's reworked.